airflow/datasets/manager.py

```diff
@@ -0,0 +1,62 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from sqlalchemy.orm.session import Session
+
+from airflow.utils.log.logging_mixin import LoggingMixin
+
+
+class DatasetEventManager(LoggingMixin):
+    """
+    A pluggable class that manages operations for dataset events.
+
+    The intent is to have one place to handle all DatasetEvent-related operations, so different
+    Airflow deployments can use plugins that broadcast dataset events to each other.
+    """
+
+    def register_dataset_change(self, *, task_instance, dataset, extra=None, session: Session) -> None:
```

Comment:
what do people think about accepting kwargs here?  perhaps that would help with future compat if we add new features / interaction capability? because if we make changes to the interface, and change the way we call the methods, maybe we'll have to inspect signature or look at interface version or something.....  @uranusjr thoughts?

on one hand adding kwargs feels like it would allow a maybe a little more freedom... at the same time... depending on the changes that we make, it might behoove us to check signature (or version) anyway to know how it will operate / what it expects 🤷

Reply:
Good call. Forward compatibility with any changes we make in future Airflow versions will help plugin writers not have to do wild things to support multiple versions of Airflow.